data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/Sink.java

```diff
@@ -20,7 +21,11 @@ public interface Sink<T extends Record<?>> {
      *
      * @param records the records to write to the sink.
      */
-    void output(Collection<T> records);
+    default void output(Collection<T> records) {
+        output(records, null);
+    }
+
+    void output(Collection<T> records, PipelineIf failurePipeline);
```

Comment:
Should we have a `setFailurePipeline` method instead? I suspect the sinks are going to need to hang on this to in order to use it later in the code.

Reply:
Also, rather than passing a `PipelineIf`, I'd probably pass this to the sinks.

```
interface FailurePipeline {
  void writeAll(Collection<T> records);
}
```

All the sinks care about is writing to the pipeline. I don't think they care about the `Source` object itself. Also, the `Source` interface doesn't support writing. 

I'd even think that the implementation of `FailurePipeline` would write directly to the `Buffer`.